traitlets/traitlets.py

```diff
@@ -418,6 +420,10 @@ def _dynamic_default_callable(self, obj):
         return getattr(self, 'make_dynamic_default', None)
 
     def instance_init(self, obj):
+        if not hasattr(obj, '_trait_values'):
```

Comment:
It seems redundant to make this check on the first, and all subsequent descriptors that require a particular attribute. Maybe there should be a descriptor class method that's called once per type of descriptor attached to an instance?